### PR TITLE
[iOS][CI] Add unit test for metal on x86_64 macos machines

### PR DESCRIPTION
**Pull request by xta0**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#68236 [iOS][CI] Add unit test for metal on x86_64 macos machines**

[iOS][CI] Add unit test for metal on x86_64 macos machines

### CODE REVIEW

File: .circleci/cimodel/data/simple/ios_definitions.py [hunk 1]
```diff
@@ -70,6 +70,9 @@ def gen_tree(self):
     IOSJob(XCODE_VERSION, ArchVariant("arm64", "metal"), extra_props={
         "use_metal": miniutils.quote(str(int(True))),
         "lite_interpreter": miniutils.quote(str(int(True)))}),
+    IOSJob(XCODE_VERSION, ArchVariant("x86_64", "metal"), extra_props={
```
Comment (xta0) [inline .circleci/cimodel/data/simple/ios_definitions.py:73]: @seemethere I added a new test here, but it looks like circle didnt generate the test in config.yaml. Anything I'm missing? I did run the `regenerate.sh`.

Comment (seemethere) [reply .circleci/cimodel/data/simple/ios_definitions.py:73]: macOS workflows were migrated to GHA following https://github.com/pytorch/pytorch/pull/67717, you'll need to implement this in the GHA generation script here: https://github.com/pytorch/pytorch/blob/be281fc597485929c406515a95fac2e2fc168528/.github/scripts/generate_ci_workflows.py#L624-L654

Comment (xta0) [reply .circleci/cimodel/data/simple/ios_definitions.py:73]: @seemethere Sorry about the confusion. This is not a macOS test. It's an iOS simulator test that runs on mac machines.

Comment (seemethere) [reply .circleci/cimodel/data/simple/ios_definitions.py:73]: Oh, iOS workflows were also migrated in https://github.com/pytorch/pytorch/pull/67645, you can add new iOS workflows here:

https://github.com/pytorch/pytorch/blob/43874d79e72c041c9bcf09cfcea6cccc94f44ce7/.github/scripts/generate_ci_workflows.py#L549-L622

Comment (xta0) [reply .circleci/cimodel/data/simple/ios_definitions.py:73]: @seemethere could you explain more on how to add the test? I added the following code and ran the regenerate.sh, but nothing happens. What's the new workflow?

```
    CIWorkflow(
        arch="macos",
        build_environment="ios-12-5-1-x86-64-metal",
        test_runner_type=MACOS_TEST_RUNNER_10_15,
        exclude_test=True,
        ciflow_config=CIFlowConfig(
            labels={LABEL_CIFLOW_IOS, LABEL_CIFLOW_MACOS},
        ),
    ),
 ```

**Pull request closed without merging**